### DISCUSSION TITLE
editable为false时，如何触发验证？

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#2244] [Q&A 2.x] [answered]
editable为false，且始终未进入过编辑状态。未进入编辑状态，则无此字段状态，那么此时该如何验证该字段。

场景：
表格组件中每一列中使用了`Editable`进行编辑状态控制，并设置了`required`等验证规则，如果未进入编辑状态，
提交时无法对该字段进行验证。

目前想到的方式是在表格`onInputValuesChange`时，对表格所有数据进行一次验证，并将验证信息存储到表格本身的`selfErrors`中。
但是表格本身的`selfErrors`无法精确记录到某行某列的具体错误（精确记录以便表格列给出对应的提示文案）。

期望：
1. Formilly是否应该提供editable为false时的官方验证方案
2. setSelfErrors是否支持自定义错误信息格式，目前message类型只能为字符串

**Comment by janryWang** [comment, marked as answer]
1. 阅读态代表，可阅读的数据就是合法数据，所以不可能要支持校验
2. errors格式你可以通过表达式定制内容

**Reply by NexxLuo** [reply]
这里场景有点特别，表单本身存在草稿状态和可提交状态，草稿状态实际上是不需要任何验证的。当第二次打开表单时，表格没有进入编辑状态，这时如果要保存表单为可提交状态，此时就需要验证了。
另外，不知道field的validate方法能否重写或者自定义，这样在form.validate或form.submit时就能自动触发了。如果没有的话，用field.setValidator也能实现
